### PR TITLE
PyTorch_NCF fix, normalize running loss

### PR DESCRIPTION
**Pull request by hieuddo**
### Description

Minor fix for PyTorch_NCF, `nn.BCELoss(reduction="sum")`, default is `reduction="mean"`

### CODE REVIEW

**Comment by tqtg** [comment]
I see your point. Two perspectives:

1. We want to show what's the current loss value on the training dataset which is what the formula says if we're about to do batch learning, although it's not quite what the model is optimizing for when doing mini-batch.
2. Normalizing the loss by the size of mini-batch will help learning be more stable, less sensitive to the size of mini-batch. Think of them as doing SGD using the `centroid` of mini-batch.

I would suggest to keep the `reduction="mean"`. If you're not comfortable with the current way of showing the loss value, we can change `count` to counting the number of mini-batches, and `sum_loss` to summing all the normalized mini-batch losses.

**Comment by tqtg** [comment]
Let's also synchronize this change to the tensorflow backend?

**Review by tqtg — APPROVED** [review]
